test/functional/apps/bundles/index.js

```diff
@@ -18,6 +18,15 @@ export default function ({ getService }) {
 
     let buildNum;
     before(async () => {
+      // Wait for status to become green
+      let status;
+      const start = Date.now();
+      do {
+        const resp = await supertest.get('/api/status');
+        status = resp.status;
+        // Stop polling once status stabilizes OR once 40s has passed
+      } while (status !== 200 && Date.now() - start < 40_000);
+
```

Comment:
This kind of logic is going to be more and more necessary in some tests. We should either:
1) Create a centralized helper service for this like `kibana.waitForAvailable()`; OR
2) Change how `@kbn/test` waits for Kibana to start. Currently it waits for a log message that the the server is ready. We could either add logic for polling the API or add a log message for when Kibana actually turns to available ("green") for the first time.

Reply:
> We could either add logic for polling the API or add a log message for when Kibana actually turns to available ("green") for the first time.

IIRC @pgayvallet has encountered this problem before. Hasn't it been fixed?

Reply:
>  We could either add logic for polling the API or add a log message for when Kibana actually turns to available ("green") for the first time

This is what I tried in https://github.com/elastic/kibana/pull/92568. Got reverted because it added a very significant time to the FTR CI tests. But maybe now that this PR fixes these delays, it would be better?

Reply:
I'll explore this as a follow up and see how long much it increase test time. I want to keep this PR clean enough to backport to 7.13.3

Reply:
Opened a new PR for this along with an additional 5s improvement on the initial `available`: https://github.com/elastic/kibana/pull/102108